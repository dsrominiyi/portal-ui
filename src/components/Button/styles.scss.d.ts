export const button: string;
export const success: string;
export const primary: string;
export const secondary: string;
export const warning: string;
export const danger: string;
