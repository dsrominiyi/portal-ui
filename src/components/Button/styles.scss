@import '../../style/_vars';

.button {
  margin: 0;
  overflow: visible;
  display: inline-block;
  padding: 10px 30px;
  outline: none;
  vertical-align: middle;
  font-size: 0.875rem;
  text-align: center;
  text-transform: uppercase;
  background-color: transparent;
  color: $brand-heavy-thunder;
  border: 1px solid $brand-off-white;
  transition: 0.1s ease-in-out;
  transition-property: color, background-color, border-color;
  cursor: pointer;

  &:disabled {
    background-color: $brand-white-smoke;
    color: $brand-grey-white;
    cursor: default;
  }

  &:hover:not(:disabled) {
    text-decoration: none;
    border: 1px solid darken($brand-off-white, 15%);
  }
}

.success {
  background-color: $brand-success;
  color: $brand-white;
  border: 1px solid transparent;

  &:hover:not(:disabled) {
    text-decoration: none;
    background-color: lighten($brand-success, 10%);
  }
}

.primary {
  background-color: $brand-primary;
  color: $brand-white;
  border: 1px solid transparent;

  &:hover:not(:disabled) {
    text-decoration: none;
    background-color: lighten($brand-primary, 10%);
  }
}

.secondary {
  background-color: $brand-secondary;
  color: $brand-black;
  border: 1px solid transparent;

  &:hover:not(:disabled) {
    text-decoration: none;
    background-color: lighten($brand-secondary, 10%);
  }
}

.warning {
  background-color: $brand-warning;
  color: $brand-white;
  border: 1px solid transparent;

  &:hover:not(:disabled) {
    text-decoration: none;
    background-color: lighten($brand-warning, 10%);
  }
}

.danger {
  background-color: $brand-danger;
  color: $brand-white;
  border: 1px solid transparent;

  &:hover:not(:disabled) {
    text-decoration: none;
    background-color: lighten($brand-danger, 10%);
  }
}
