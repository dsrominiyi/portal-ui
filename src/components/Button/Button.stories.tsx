import React from 'react';
import { storiesOf } from '@storybook/react';

import Button from '.';
import styles from '../../containers/Login/styles.scss';
import buttonStyles from './styles.scss';

const stories = storiesOf('Button', module);

const render = (
  label: string,
  status?: Status,
  classNames?: string[],
  additionalClasses?: string[],
  isDisabled?: boolean
) => () => (
  <Button
    label={label}
    status={status}
    classNames={classNames}
    additionalClasses={additionalClasses}
    onClick={() => alert('You clicked!')}
    isDisabled={isDisabled}
  />
);

export const storyRenders = {
  default: render('TEST'),
  extraClass: render('TEST', undefined, undefined, [styles.loginButton]),
  customStyling: render('TEST', undefined, ['styles.customButton']),
  primary: render('TEST', undefined, undefined, [buttonStyles.primary]),
  secondary: render('TEST', undefined, undefined, [buttonStyles.secondary]),
  success: render('TEST', undefined, undefined, [buttonStyles.success]),
  warning: render('TEST', undefined, undefined, [buttonStyles.warning]),
  danger: render('TEST', undefined, undefined, [buttonStyles.danger]),
  disabled: render('TEST', undefined, undefined, undefined, true)
};

stories.add('Default', storyRenders.default);
stories.add('Additional class', storyRenders.extraClass);
stories.add('Custom styling', storyRenders.customStyling);
stories.add('Primary button', storyRenders.primary);
stories.add('Secondary button', storyRenders.secondary);
stories.add('Success button', storyRenders.success);
stories.add('Warning button', storyRenders.warning);
stories.add('Danger button', storyRenders.danger);
stories.add('Disabled', storyRenders.disabled);
