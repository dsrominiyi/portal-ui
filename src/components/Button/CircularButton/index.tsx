import React, { FunctionComponent, MouseEvent, ReactNode } from 'react';
import Button from '../index';

import styles from './styles.scss';

interface Props {
  label: ReactNode;
  onClick(event: MouseEvent): void;
}

export const CircularButton: FunctionComponent<Props> = ({ label, onClick }: Props) => (
  <Button label={label} classNames={[styles.circularButton]} onClick={onClick} />
);

CircularButton.displayName = 'CircularButton';
export default CircularButton;
