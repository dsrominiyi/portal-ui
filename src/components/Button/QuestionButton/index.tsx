import React, { FunctionComponent, MouseEvent } from 'react';
import classNames from 'classnames/bind';
import Button from '../index';
import styles from './styles.scss';

interface Props {
  label: string;
  isActive?: boolean;
  isDisabled?: boolean;
  attributes?: { [attr: string]: string };
  onClick(event: MouseEvent): void;
}

const cx = classNames.bind(styles);

export const QuestionButton: FunctionComponent<Props> = ({
  label,
  isActive,
  isDisabled,
  attributes,
  onClick
}: Props) => (
  <Button
    label={label}
    classNames={[cx({ questionButton: true, active: isActive })]}
    onClick={onClick}
    isDisabled={isDisabled}
    attributes={attributes}
  />
);

QuestionButton.displayName = 'QuestionButton';
export default QuestionButton;
