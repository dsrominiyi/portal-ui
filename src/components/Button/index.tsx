import React, { FunctionComponent, MouseEvent, ReactNode } from 'react';
import classnames from 'classnames/bind';

import statusClassMapper from '../../helpers/statusClassMapper';

import styles from './styles.scss';

interface Props {
  label: ReactNode;
  status?: Status;
  classNames?: string[];
  additionalClasses?: string[];
  isDisabled?: boolean;
  attributes?: { [attr: string]: string };
  onClick?(event: MouseEvent): void;
}

const cx = classnames.bind(styles);

export const Button: FunctionComponent<Props> = ({
  label,
  onClick,
  status,
  classNames,
  additionalClasses = [],
  isDisabled,
  attributes = {}
}: Props) => {
  const buttonClasses = classNames
    ? classNames.join(' ')
    : [styles.button, ...additionalClasses, cx({ ...statusClassMapper(status) })].join(' ');

  return (
    <button className={buttonClasses} onClick={onClick} disabled={isDisabled} {...attributes}>
      {label}
    </button>
  );
};

Button.displayName = 'Button';
export default Button;
