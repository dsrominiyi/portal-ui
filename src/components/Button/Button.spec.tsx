import { storyRenders } from './Button.stories';

import testStorySnapshots from '../../helpers/test/testStorySnapshots';

describe('Button', () => {
  test('component story renders match snapshots', () => {
    testStorySnapshots(storyRenders);
  });
});
