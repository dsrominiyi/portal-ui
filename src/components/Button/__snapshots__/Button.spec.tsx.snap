// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component story renders match snapshots: customStyling 1`] = `
<button
  className="styles.customButton"
  onClick={[Function]}
>
  TEST
</button>
`;

exports[`Button component story renders match snapshots: danger 1`] = `
<button
  className="button danger "
  onClick={[Function]}
>
  TEST
</button>
`;

exports[`Button component story renders match snapshots: default 1`] = `
<button
  className="button "
  onClick={[Function]}
>
  TEST
</button>
`;

exports[`Button component story renders match snapshots: disabled 1`] = `
<button
  className="button "
  disabled={true}
  onClick={[Function]}
>
  TEST
</button>
`;

exports[`Button component story renders match snapshots: extraClass 1`] = `
<button
  className="button loginButton "
  onClick={[Function]}
>
  TEST
</button>
`;

exports[`Button component story renders match snapshots: primary 1`] = `
<button
  className="button primary "
  onClick={[Function]}
>
  TEST
</button>
`;

exports[`Button component story renders match snapshots: secondary 1`] = `
<button
  className="button secondary "
  onClick={[Function]}
>
  TEST
</button>
`;

exports[`Button component story renders match snapshots: success 1`] = `
<button
  className="button success "
  onClick={[Function]}
>
  TEST
</button>
`;

exports[`Button component story renders match snapshots: warning 1`] = `
<button
  className="button warning "
  onClick={[Function]}
>
  TEST
</button>
`;
